doc/content/components/network-server.md

```diff
@@ -7,3 +7,51 @@ weight: 3
 The Network Server handles the LoRaWAN network layer, including MAC commands, regional parameters and adaptive data rate (ADR).
 
 <!--more-->
+
+## Device Management
+
+Network Servers expose `Get`, `Set` and `Delete` gRPC methods for device management. REST analogues for these methods exist as well. Typical clients of these RPC methods are [Console]({{< ref "/components/console.md" >}}) and [CLI]({{< ref "/components/cli.md" >}}).
+
+In case the device has an active session and it is in Class B or C, a downlink task may added to internal downlink task queue following the device update.
+
+## Application Downlink Queue Management
+
+Network Servers expose `DownlinkQueueReplace`, `DownlinkQueuePush` and `DownlinkQueueList` gRPC methods for application downlink management. Typical clients of these RPC methods are [Application Servers]({{< ref "/components/application-server.md" >}}).
+
+In case the device has an active session and it is in Class B or C, a downlink task may added to internal downlink task queue following the application queue update.
+
+## Application Linking
+
+Network Servers expose `LinkApplication` gRPC method for application linking. Typical clients of this RPC method are [Application Servers]({{< ref "/components/application-server.md" >}}).
+
+Once the link is established, Network Server will send application-specific `ApplicationUp` messages to the client via the link. There can be at most one active link per-appplication.
+
+In case link is not active, but Network Server has application-specific `ApplicationUp` messages to send, those messages will be queued and sent once the link is established.
+
+## Downlink Scheduling
+
+Network Server maintains internal downlink task queue. Each downlink task has an execution time associated with it, which tasks are sorted by in ascending order. Whenever a downlink task is ready to execute, it is executed as soon as possible.
+
+As first step of scheduling a downlink, Network Server attempts to construct a `DownlinkMessage`.
+
+In case a pending session exists and Join-Accept is queued for the device, `DownlinkMessage` will contain the Join-Accept. Otherwise, Network Server attempts to genererate a data downlink in the active session, which `DownlinkMessage` will contain.
+
+If Network Server successfully constructed a `DownlinkMessage`, it is passed to Gateway Server to schedule via `ScheduleDownlink`.
+
+## Uplink Handling
+
+Network Servers expose `HandleUplink` gRPC method, which is used to receive device uplinks. Typical clients of this RPC method are [Gateway Servers]({{< ref "/components/gateway-server.md" >}}).
+
+Network Servers process the uplinks received and handle accordingly. The first step is matching the uplink to a device. In case an uplink cannot be matched to a device stored in Network Server, it is dropped.
+
+- If Join-Request is received:
```

Comment:
Avoid bullet lists in documentation

Reply:
Join-request no title case

Reply:
Indeed use small headings here and a numbered list to show order